dotenv/src/lib.rs

```diff
@@ -20,6 +20,16 @@ use crate::find::Finder;
 pub use crate::iter::Iter;
 
 static START: Once = Once::new();
+static FILES: &[&'static str] = &[
+    ".env",
+    ".env.production",
+    ".env.test",
+    ".env.development",
+    ".env.local",
+    ".env.production.local",
+    ".env.test.local",
+    ".env.development.local",
+];
```

Comment:
You may want to consider the `dotenv-lookup` approach:

https://github.com/dotenv-linter/dotenv-linter/blob/6f995219c6847087db8a80d01048bda569cd96e0/dotenv-lookup/src/file.rs#L9-L11

https://github.com/dotenv-linter/dotenv-linter/blob/6f995219c6847087db8a80d01048bda569cd96e0/dotenv-lookup/src/file.rs#L78-L85

They additionally use that method [to do a recursive lookup](https://github.com/dotenv-linter/dotenv-linter/blob/6f995219c6847087db8a80d01048bda569cd96e0/dotenv-lookup/src/lib.rs#L63-L96) (_for linting reasons_).

I mention this as hard-coded filenames like this probably isn't wise to maintain, you'd be better off with the pattern. Consider a user has a different env filename variation for `staging` or something more niche.